src/representor/jsonapi.ts

```diff
@@ -8,15 +8,35 @@ import Representation from './base';
  */
 type JsonApiLink = string | { href: string };
 
+/**
+ * This type is a full 'links' object, which might appear on the top level
+ * or on resource objects.
+ */
+type JsonApiLinksObject = {
+  self?: JsonApiLink,
+  profile?: JsonApiLink
+  [rel: string]: JsonApiLink | JsonApiLink[]
+};
+
+/**
+ * This is a single JSON:API resource. Its type contains just the properties
+ * we care about.
+ */
+type JsonApiResource = {
+  type: string,
+  id: string,
+  links?: JsonApiLinksObject,
+};
+
+
 /**
  * This type represents a valid JSON:API response. We're only interested
  * in the links object at the moment, so everything else is (for now)
  * untyped.
  */
 type JsonApiObject = {
```

Comment:
This should be renamed to `TopLevelObject`. Sorry, I should have spotted that before. There's already a [JSON:API Object](https://jsonapi.org/format/#document-jsonapi-object) that looks different than this.

Reply:
Fixed